x-pack/plugins/cases/public/components/all_cases/assignees_filter.tsx

```diff
@@ -33,6 +35,8 @@ const AssigneesFilterPopoverComponent: React.FC<AssigneesFilterPopoverProps> = (
   setFetchAssignees,
 }) => {
   const { owner: owners } = useCasesContext();
+  const hasOwners = owners.length > 0;
+  const availableOwners = useAvailableCasesOwners(['read']);
```

Comment:
When using the modal for cases owners can be an empty array so we'll only fetch the assignees that the user has read access to.